workspaceitem-data-upload.md

```diff
@@ -0,0 +1,205 @@
+# WorkspaceItem data of upload sectionType
+[Back to the definition of the workspaceitems endpoint](workspaceitems.md)
+
+The section data represent the data about the user uploaded files
+
+```json
+{
+	"files": [ 
+  	 	{
+  	 		metadata: {
+  	 			"dc.title" : [{value: "sample_file.pdf"}],
+  	 			"dc.description" : [{value: "Description of the sample file"}]
+  	 		},
+  	 		"sizeBytes": 8528,
+			"checkSum": {
+			    "checkSumAlgorithm": "MD5",
+			    "value": "9d8f0f9e369cf12159d47c146c499cf4"
+			},
+  	 		"url": "http://dspace7.4science.it/api/core/bitstreams/00001abf-b2e0-477a-99de-104db7cb6469/content",
+  	 		"accessConditions": [
+  	 			{
+  	 				"id": 123,
+	  	 			"name": "openaccess",
+	  	 			"groupUUID": "uuid-of-the-anonymous-group"
+  	 			},
+  	 			{
+  	 				"id": 126,
+	  	 			"name": "administrator",
+	  	 			"groupUUID": "uuid-of-the-administrator-group"
+  	 			},
+  	 			{
+  	 				"id": 127,
+	  	 			"name": "embargo",
+	  	 			"groupUUID": "1faf7c51-2a14-4826-b0b1-f1c1d2d82dd7",
+	  	 			"startDate": "2018-06-24T00:40:54.970+0000"
+  	 			},
+  	 			{
+  	 				"id": 128,
+	  	 			"name": "lease",
+	  	 			"groupUUID": "38ecd5ae-af12-4144-a276-81532e1679f8",
+	  	 			"endDate": "2017-12-24T00:40:54.970+0000"
+  	 			}
+  	 		]
+ 		}
+ 	]
+}
+```
+the files attribute contains the list of user uploaded file in the section. For each file the following attributes exist
+* metadata: the map of the metadata assigned to the specific file with [the same structure](workspaceitem-data-metadata.md) used in the submission-form sectionType for the item metadata
+* sizeBytes (**READ-ONLY**): the size of the received file as calculated on the server at the receiving time 
+* checkSum (**READ-ONLY**): the checksum details (algorithm and value) of the received file as calculated on the server at the receiving time
+* url (**READ-ONLY**): from where the file can be downloaded
+* accessConditions: an array of all the policies that has been applied by the user to the file. 
+
+## Patch operations
+The PATCH method expects a JSON body according to the [JSON Patch specification RFC6902](https://tools.ietf.org/html/rfc6902)
+
+Each successful Patch operation will return a HTTP 200 CODE with the new workspaceitem as body. See also the [General rules for the Patch operation](patch.md) for more details.
+
+### Add
+The add operation is used to add new metadata or access condition to already uploaded files. To upload a completely new file a [POST Multipart request must be issued against the workspaceitems endpoint, see here](workspaceitems.md).
+
+#### Metadata
+To add a value to an **existent metadata** the client must send a JSON Patch ADD operation as follow
+
+`curl --data '{[ { "op": "add", "path": "/sections/<:name-of-the-form>/files/<:file-idx>/metadata/<:metadata>/-", "value": {value: "...", authority: "...", confidence: 600, language: ".."}}]}' -X PATCH ${dspace7-url}/api/submission/workspaceitems/<:id>`
```

Comment:
At some point, we will also need to implement these operations for the `/api/core/bitstreams/<:uid>` endpoint. I have the feeling that we'll be duplicating this functionality then.

Reply:
This is similar to the `/api/core/items/<:uid>` using a patch request against these urls is an administrative edit that should require higher privileges (ADMIN) to be performed and can force everything ignoring validation / configuration on the submission similary to what an administrator can do in the current DSpace with the edit functionality.
Please note that I this that we should also allow the ADMIN to use the same "submission UI" to edit existent item, as BTW it is already implemented in DSpace-CRIS, but this should be an additional feature and not a replacement of the administrative edit